src/helpers/check-merge-safety.ts

```diff
@@ -44,13 +45,26 @@ export const checkMergeSafety = async (inputs: CheckMergeSafety) => {
 
 const setMergeSafetyStatus = async (pullRequest: PullRequest, { context = 'Merge Safety', ...inputs }: CheckMergeSafety) => {
   const { state, message } = await getMergeSafetyStateAndMessage(pullRequest, inputs);
-  await setCommitStatus({
-    sha: pullRequest.head.sha,
-    state,
-    context,
-    description: message,
-    ...githubContext.repo
-  });
+  const hasExistingFailureStatus = await checkForExistingFailureStatus(pullRequest, context);
+  if (hasExistingFailureStatus && state === 'failure') {
```

Comment:
Isn't this `state === 'failure'` check already happening inside `checkForExistingFailureStatus`?

Reply:
`state` here is the current status, whereas the value from `checkForExistingFailureStatus` is coming from the previous status

basically added this check in to say that in the case where there is an existing `failure` status, only skip if the new status is also a `failure` (it's almost impossible for the status to flip from from `failure` to `success` unless someone were to force-push the target branch, just wanted to make sure our bases were covered)

Reply:
Gotcha - I was getting confused between this state and the individual PR "combined" state